### PR TITLE
Fix train-minimal feature and ensure it gets tested

### PR DESCRIPTION
**Pull request by dae**
### Checklist

- [x] Confirm that `run-checks` script has been executed.

### CODE REVIEW

File: burn-train/src/metric/dashboard/cli_stub.rs [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 /// A simple renderer for when the cli feature is not enabled.
 pub struct CLIDashboardRenderer;
 
 impl CLIDashboardRenderer {
     /// Create a new instance.
-    pub fn new() -> Self {
+    pub fn new(_interuptor: TrainingInterrupter) -> Self {
         Self {}
     }
```
Comment (nathanielsimard) [inline burn-train/src/metric/dashboard/cli_stub.rs:13]: I think the problem is that the renderer type is selected with feature flag over the type, where I think it should be when creating. I think we should have a function like the following:

```rust
pub fn rendered_default(interuptor: TrainingInterrupter) -> Rendered {
}
```

File: examples/guide/Cargo.toml [hunk 2]
```diff
@@ -6,8 +6,13 @@ name = "guide"
 publish = false
 version = "0.10.0"
 
+[features]
+default = ["train"]
+train = ["burn/train"]
+train-minimal = ["burn/train-minimal"]
+
 [dependencies]
-burn = {path = "../../burn", features = ["train", "wgpu"]}
```
Comment (nathanielsimard) [inline examples/guide/Cargo.toml:10]: I think the guide needs the `train` feature flag!

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
I think we could improve where the renderer is created to avoid mistakes like this!

File: examples/guide/Cargo.toml (hunk 2 above)
Comment (dae) [reply examples/guide/Cargo.toml:10]: It's included by the default features above. If you'd prefer not to have the extra complexity in the guide, can you suggest a simple other model this example could depend on? I based the example off of your custom-training-loop one.

**Review by nathanielsimard — APPROVED** [review]
Thanks for the fix! I think it's cleaner now!

**Pull request merged**